modules/ROOT/pages/munit-matchers.adoc

```diff
@@ -5,251 +5,372 @@ endif::[]
+----
+#[MunitTools::either(MunitTools::nullValue(),MunitTools::equalTo(0))]
+----
+
+* xref:core-matchers-reference.adoc#not-matcher[not(Matcher)] checks if the provided matcher is not successful.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::not(MunitTools::equalTo(0))]
+----
+* xref:core-matchers-reference.adoc#anyof-matchers[anyOf(Matchers[\])] checks if any of the matchers are successful.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::anyOf(MunitTools::notNullValue(),MunitTools::withMediaType('text/xml'),MunitTools::isEmptyString())]
+----
+
+* xref:core-matchers-reference.adoc#allof-matchers[allOf(Matchers[\])] Checks if all of the matchers are successful.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::allOf(MunitTools::notNullValue(),MunitTools::withMediaType('text/xml'),MunitTools::isEmptyString())]
+----
+
+== Match Strings
+
+String Matchers perform assertions on String expressions:
+
+* Does the expression contain a specified string?
+* Does the expression start or end with a specified string?
+* Does the string in the expression have zero length?
+* Does the string in the expression match a specified string?
+
+String Matchers:
+
+* xref:string-matchers-reference.adoc#containsstring-string[containsString(String)] checks that the expression contains the specified String.
+++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::containsString('example')]
+----
+
+* xref:string-matchers-reference.adoc#startswith-string[startsWith(String)] checks that the expression starts with the specified String.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::startsWith('exam')]
+----
+
+* xref:string-matchers-reference.adoc#endswith-string[endsWith(String)] checks that the expression ends with the specified String.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::endsWith('ple')]
+----
+
+* xref:string-matchers-reference.adoc#isemptystring[isEmptyString()] checks that the expression has zero length.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::isEmptyString()]
+----
+
+* xref:string-matchers-reference.adoc#isemptyornullstring[isEmptyOrNullString()] checks that the expression is null, or has zero length.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::isEmptyOrNullString()]
+----
+
+* xref:string-matchers-reference.adoc#equaltoignoringcase-string[equalToIgnoringCase(String)] checks that the expression is equal to the specified String, ignoring case.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::equalToIgnoringCase('Example')]
+----
+
+* xref:string-matchers-reference.adoc#equaltoignoringwhitespace-string[equalToIgnoringWhiteSpace(String)] checks that the expression is equal to the string disregarding leading and trailing white spaces, and compression all inner white spaces to a single space.
```

Comment:
Not sure about this one.  

checks that the expression is equal to the string, disregarding leading and trailing white spaces, and **compressing** all inner white spaces to a single space. 

?